code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -110,37 +110,23 @@
 /obj/item/weapon/storage/lockbox/emp_act(severity)
 	..()
 	if(!broken)
+		var/probab
 		switch(severity)
 			if(1)
-				if(prob(80))
-					locked = !locked
-					src.update_icon()
-					if(!locked)
-						for(var/atom/movable/A in src)
-							for(var/obj/O in src)
-							remove_from_storage(A, get_turf(src))
-						if(oneuse)
-							qdel(src)
+				probab = 80
 			if(2)
-				if(prob(50))
-					locked = !locked
-					src.update_icon()
-					if(!locked)
-						for(var/atom/movable/A in src)
-							for(var/obj/O in src)
-							remove_from_storage(A, get_turf(src))
-						if(oneuse)
-							qdel(src)
-			if(3)
-				if(prob(25))
-					locked = !locked
-					src.update_icon()
-					if(!locked)
-						for(var/atom/movable/A in src)
-							for(var/obj/O in src)
-							remove_from_storage(A, get_turf(src))
-						if(oneuse)
-							qdel(src)
+				probab = 50
+		if(prob(probab))
+			locked = !locked
+			src.update_icon()
+			if(!locked)
+				for(var/atom/movable/A in src)
+					for(var/obj/O in src)
```

Comment:
I know this is (almost) what the old code did but this is unnecessary. It's looping through all movables, and for each movable found it's looping AGAIN through all objects in src. You can just remove this line and de-indent the one after it.